server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1799,6 +1801,18 @@ boolean shouldRollTranslogGeneration() {
     }
 
     public void onSettingsChanged() {
+        var newEngineFactory = getEngineFactory();
+        if (newEngineFactory.getClass() != engineFactory.getClass()) {
+            try {
+                // resetEngineToGlobalCheckpoint will use the new engine factory
+                indexShardOperationPermits.blockOperations(30, TimeUnit.MINUTES, this::resetEngineToGlobalCheckpoint);
+            } catch (InterruptedException | TimeoutException ignored) {
+                logger.warn("Waiting to block in-flight operations failed, could not switch to new engine.");
+            } catch (IOException e) {
+                logger.warn("Exception raised while trying to switch to new engine", e);
```

Comment:
Is this fine to just warn and continue? Shouldn't the shard fail or something?

Reply:
Hm right, I only had our concrete LR engine switch in mind were I thought it's better to leave the shard with a read-only engine instead of failing it. But as a general concept, failing it sounds safer.